Implement RUDPNetworkInterface

IgorFilipovich opened the pull request:
Implement NetworkInterface using reliable UDP
RUDPNetworkInterfaceImpl starts udp server on passed address and send messages
from the same address

Derovi committed:
UserManager signature

Derovi committed:
Merge branch 'develop' into managers

Derovi committed:
work on NetworkService

Derovi committed:
NetworkService waitForAMessage method

Derovi committed:
UserManager ::sendNetworkMessage

Derovi committed:
Data manager

Derovi committed:
Last IP saving for contacts

Derovi committed:
Fixed bug in waitForAMessage

Derovi committed:
Dismess-protocol

Derovi committed:
Dismess-protocol implementation

Derovi committed:
Fixed WaitForAMessage used requestHandlers

Derovi committed:
Sender in responses now InetSocketAddress

Derovi committed:
Sending approves

Derovi committed:
NetworkService responses

Derovi committed:
NetworkService bug fix, tests

Derovi committed:
Fixed codestyle

Derovi committed:
Merge pull request #13 from Derovi/managers

NetworkService (L7 Dismess-Protocol implementation)

IgorFilipovich committed:
Start SecureNetworkInterface (#12)

* Start SecurityNetworkInterface

Add ecnryption with use of AES256 symmetric ecnryption and DiffieHellman
assymetric encryption.
Implement Dismess Protocol.
Add unit-tests for Encryptor and SecureNetworkInterface
Finish SecureNetworkInterface with session management

IgorFilipovich committed:
Implement RUDPNetworkInterface

Implement NetworkInterface using reliable UDP
RUDPNetworkInterfaceImpl starts udp server on passed address and send messages
from the same address

VKuzia commented on src/test/kotlin/by/dismess/core/network/RUDPNetworkInterfaceTest.kt:
```diff
@@ -0,0 +1,166 @@
+                user.start(sockets[i])
+                user.setMessageReceiver { sender: InetSocketAddress, data: ByteArray ->
+                    var from = 0
+                    for (j in 0 until sockets.size) {
```
from было бы приятнее сделать более говорящей, откуда или что берётся...

IgorFilipovich replied:
fixed

VKuzia commented on src/test/kotlin/by/dismess/core/network/RUDPNetworkInterfaceTest.kt:
```diff
@@ -0,0 +1,166 @@
+                    messages[from][i].removeAt(0)
+                    Assert.assertArrayEquals(message, data)
+                    ++receiveCounter
+                    Unit
```
это возвращаемое значение? оно тут обязательно?

IgorFilipovich replied:
Иначе ломается лямбда, либо так, либо делать receiveCounter += 1

VKuzia replied:
ок

VKuzia reviewed (approved):
Если верить либе, то, вроде как, всё хорошо

IgorFilipovich committed:
Gradle fix
Change naming in test

IgorFilipovich committed:
Style fix

Derovi committed:
Chating system (#17)

* UserManager requests

* UserManager::ping

* UserManager

* UserManager retrieveUser, avatar

* Base chat system

* ChatManager

* New chating

* ChunkID, FlowID, MessageID

* Documentation

* Message iterator

* Flow iterator

* Flow iterator creating

* Flow

* Changed Storable interface

* Send message

* Send message with coroutines

* Fixed bug with accept, persist in Flow

* ChatManager persist, accept

* Changed version to 0.4-snapshot

* Fixed errors

* Fixed ktlint

* ktlint fix

IgorFilipovich committed:
Revert "Chating system (#17)" (#20)

This reverts commit a076f5459e33c761124030afd39abc443f757257.

VKuzia committed:
Invite (#16)

Add JSTUN public public socket address getter
Add invite functions to (de)cipher invite
Add invite tests

IgorFilipovich committed:
Merge branch 'develop' into rudp-network-interface
